BatchProcessing/BatchStructureSetConversion.py

```diff
@@ -321,16 +321,20 @@ def save_rtslices(output_dir, use_ref_image, ref_image_node_id=None):
     if exist_db:
       logging.info('BatchStructureSet running in existing database mode')
       DICOMUtils.openDatabase(input_folder)
-      all_patients = slicer.dicomDatabase.patients()
+      db = slicer.dicomDatabase
+      all_patients = db.patients()
       logging.info('Must Process Patients %s' % len(all_patients))
-
       for patient in all_patients:
-        slicer.mrmlScene.Clear(0) # clear the scene
-        DICOMUtils.loadPatientByUID(patient)
-        output_dir = os.path.join(output_folder,patient)
+        slicer.mrmlScene.Clear(0) # clear the scen
+        studyList=db.studiesForPatient(patient) # code modified from https://www.slicer.org/wiki/Documentation/Nightly/ScriptRepository 

```

Comment:
This is more complicated than necessary. I think this would work better:
`slicer.dicomDatabase.fieldForPatient('PatientID', patient)`

In addition, I think the database UID should be kept, because in DICOM the patient ID is not necessarily unique, and it is unique together with the patient name, but it is fragile to use patient name in path. With the database patient UID though, it is unique within the Slicer database. Something like this:
```
patientID = slicer.dicomDatabase.fieldForPatient('PatientID', patient)
folderName = str(patient) + '_' + patientID
```
@lassoan What do you think?

Reply:
Patient ID can be really anything (it is often set to empty during image anonymization, even though it is not valid), can be too long to be used as filename, and may contain special characters that cannot be used in filenames. So, they are definitely much more risky to use than the database's patient UID.

I would recommend to make addint patient name+id to folder name optional and sanitize the string (remove special characters, limit maximum size, etc.).